documentation/api/query/v4/producers.markdown

```diff
@@ -0,0 +1,102 @@
+---
+title: "PuppetDB 4.1: Producers endpoint"
+layout: default
+canonical: "/puppetdb/latest/api/query/v4/producers.html"
+---
+
+[curl]: ../curl.html#using-curl-from-localhost-non-sslhttp
+[paging]: ./paging.html
+[query]: ./query.html
+[subqueries]: ./ast.html#subquery-operators
+[factsets]: ./factsets.html
+[reports]: ./reports.html
+[catalogs]: ./catalogs.html
+
+Producers are the Puppet masters that send reports, catalogs, and factsets to PuppetDB.
+
+When PuppetDB stores a report, catalog, or factset, it keeps track of the producer
+of the report/catalog/factset. PuppetDB also keeps a list of producers it has seen.
+You can query this list by making an HTTP request to the `/producers` endpoint.
+
+## `/pdb/query/v4/producers`
+
+This will return all producers known to PuppetDB.
+
+### URL parameters
+
+* `query`: optional. A JSON array containing the query in prefix notation. If
+  not provided, all results will be returned. See the sections below for the
+  supported operators and fields. For general info about queries,
+  see [our guide to query structure.][query]
+
+### Query operators
+
+See [the AST query language page](./ast.html)
+
+### Query fields
+
+* `"name"` (string): the certname of a producer.
+
+### Subquery relationships
+
+The following list contains related entities that can be used to constrain the result set by using implicit subqueries. For more information, consult the documentation for [subqueries][subqueries].
+
+* [`factsets`][factsets]: factsets received for a producer.
+* [`reports`][reports]: reports received for a producer.
+* [`catalogs`][catalogs]: catalogs received for a producer.
+
+### Response format
+
+The response is a JSON array of hashes, where each hash has the form:
+
+    {"name": <string>}
+
+The array is unsorted.
```

Comment:
I'd at least be inclined to change it in the bits we're adding, not sure about the rest right now.